test/websocket.ganache.js

```diff
@@ -570,18 +565,23 @@ describe('WebsocketProvider (ganache)', function () {
 
             setTimeout(async function(){
                 assert(stage === 1);
-
+                let blockNumber;
                 const deferred = web3.eth.getBlockNumber();
 
                 server = ganache.server(ganacheOptions);
                 await server.listen(port);
-
-                const blockNumber = await deferred;
-                assert(blockNumber === 0);
-
-                web3.currentProvider.removeAllListeners();
-                resolve();
-            },2500);
+                try {
+                    blockNumber = await deferred;
+                } catch (error) {
+                    reject();
+                }
+                if (blockNumber === undefined) {
+                    reject();
+                }
+            },1000);
+        await server.close();
+        web3.currentProvider.removeAllListeners();
+        resolve();
```

Comment:
I think it would be better to resolve inside the `setTimeout`, because there is a chance to resolve before the code inside has been run and not check the conditions.

Reply:
i agree, thank you for the review